### ISSUE TITLE
[Bug] MSAL Logs are not clear when regional is enabled and tokens are acquired from the cache

### ISSUE THREAD

**Issue by trwalke** [AzureAD/microsoft-authentication-library-for-dotnet#3073]
When performing `AcquireTokenForClient()` and the region is configured, if MSAL has no tokens in the cache, MSAL will properly log the following message when acquiring the token from the identity provider:

"Fetched access token from host region.r.login.microsoftonline.com. Endpoint https://eastus.r.login.microsoftonline.com/Some Id/"

if MSAL already has a token in the cache, MSAL will log the following when returning the token from the cache:

"Fetched access token from host login.microsoftonline.com. Endpoint https://login.microsoftonline.com/Some Id/"

This can be very confusing for customers who expect all token acquisition traffic to go to the regional endpoint.

**Which version of MSAL.NET are you using?**
MSAL.NET 4.39.0

**Platform**
.NET FWK, .NET Core,

**What authentication flow has the issue?**
* Desktop / Mobile
    * [ ] Interactive
    * [ ] Integrated Windows Authentication
    * [ ] Username Password
    * [ ] Device code flow (browserless)
* Web app 
    * [ ] Authorization code
    * [ ] On-Behalf-Of
* Daemon app 
    * [x] Service to Service calls

Other?

**Is this a new or existing app?**

**Repro**

```csharp
            IConfidentialClientApplication app = ConfidentialClientApplicationBuilder.Create(clientId)
                .WithAzureRegion("SomeRegion")
                .WithCertificate(Cert)
                .Build();

            app.AcquireTokenForClient(scopes).ExecuteAsync(); //Logs are correct
            app.AcquireTokenForClient(scopes).ExecuteAsync(); //Logs are misleading
```

**Expected behavior**
It should be clear that MSAL is acquiring the token from the cache and not from the global endpoint

**Actual behavior**
"Fetched access token from host login.microsoftonline.com. Endpoint https://login.microsoftonline.com/Some Id/"

**Additional context / logs / screenshots / links to code**
Relevant migration conversation: [Conversation](https://teams.microsoft.com/l/message/19:e19672c8c1bb4237a5a3356cda5880fd@thread.tacv2/1638287735276?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=7e394638-cacd-40c7-9e80-c4434c4c90df&parentMessageId=1638287735276&teamName=.NET%20Migration%20(ADAL.NET%20to%20MSAL.NET%20and%20SAL%2FMISE)&channelName=General&createdTime=1638287735276)

**Comment by bgavrilMS** [comment]
For issues that are known to be bugs please tag as "Bug" and assign a priority. 

P0 = high impact, many users affected (e.g. a common scenario is blocked)
P1 = low impact x many users OR high impact x few users (e.g. a scenario is blocked but it is not very common)
P2 = P1 but workaround exists, supportability issues etc.
P3 = nice to have

**Comment by SameerK-MSFT** [comment]
Fixed in [MSAL 4.40.0](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.40.0)
